sonic_platform_base/chassis_base.py

```diff
@@ -451,6 +453,95 @@ def get_thermal_manager(self):
         """
         raise NotImplementedError
 
+    ##############################################
+    # Voltage Sensor Methods
+    ##############################################
+
+    def get_num_vsensors(self):
+        """
+        Retrieves the number of voltage sensors available on this chassis
+
+        Returns:
+            An integer, the number of voltage sensors available on this chassis
+        """
+        return len(self._vsensor_list)
+
+    def get_all_vsensors(self):
+        """
+        Retrieves all voltage sensors available on this chassis
+
+        Returns:
+            A list of objects derived from VsensorBase representing all voltage 
+            sensors available on this chassis
+        """
+        return self._vsensor_list
+
+    def get_vsensor(self, index):
+        """
+        Retrieves voltage sensor unit represented by (0-based) index <index>
+
+        Args:
+            index: An integer, the index (0-based) of the voltage sensor to
+            retrieve
+
+        Returns:
+            An object dervied from VsensorBase representing the specified voltage sensor
+        """
+        vsensor = None
+
+        try:
+            vsensor = self._vsensor_list[index]
+        except IndexError:
+            sys.stderr.write("Voltage sensor index {} out of range (0-{})\n".format(
+                             index, len(self._vsensor_list)-1))
+
+        return vsensor
+
+    ##############################################
+    # Current Sensor Methods
+    ##############################################
+
+    def get_num_isensors(self):
+        """
+        Retrieves the number of current sensors available on this chassis
+
+        Returns:
+            An integer, the number of current sensors available on this chassis
+        """
+        return len(self._isensor_list)
+
+    def get_all_isensors(self):
+        """
+        Retrieves all Current sensors available on this chassis
+
+        Returns:
+            A list of objects derived from IsensorBase representing all current
+            sensors available on this chassis
+        """
+        return self._isensor_list
+
+    def get_isensor(self, index):
+        """
+        Retrieves current sensor object represented by (0-based) index <index>
+
+        Args:
+            index: An integer, the index (0-based) of the current sensor to
+            retrieve
+
+        Returns:
+            An object dervied from IsensorBase representing the specified Current 
```

Comment:
Typo, 'dervied' -> 'derived'

'Current' should be lowercase.

Reply:
ack